scipy/signal/_upfirdn_apply.pyx

```diff
@@ -65,6 +65,56 @@ def _output_len(Py_ssize_t len_h,
     return need
 
 
+@cython.cdivision(True)  # faster modulo
+@cython.boundscheck(False)  # designed to stay within bounds
+@cython.wraparound(False)  # we don't use negative indexing
+def _rat(x, tol=None):
```

Comment:
For what it's worth, I wrote some python code along these lines to convert a duration (in milliseconds) into a fraction (in seconds) with the numerator and denominator limited to 16 bit values; see `_msec_to_numden` here: https://github.com/WarrenWeckesser/numpngw/blob/master/numpngw.py#L940